tests/cmd/cmd_test.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/kylelemons/godebug/diff"
+)
+
+var Fixtures = os.ExpandEnv("$GOPATH/src/github.com/kedgeproject/kedge/tests/cmd/fixtures/")
+
+func TestKedgeGenerate(t *testing.T) {
+	testCases := []struct {
+		name        string
+		command     []string
+		input       string
+		wantSuccess bool
+		error       string
+	}{
+		{
+			name:        "multiple configmaps given and name not specified",
+			command:     []string{"generate", "-f=" + Fixtures + "multi-configmapname/app.yaml"},
+			wantSuccess: false,
+			error:       "unable to perform controller operations: unable to fix data: unable to fix ControllerFields: unable to fix configMaps: please specify name for app.configMaps[1]",
+		},
+		{
+			name:        "generating deploymentconfig",
+			command:     []string{"generate", "-f=" + Fixtures + "controllers/input"},
+			input:       Fixtures + "controllers/os.yaml",
+			wantSuccess: true,
+		},
+		{
+			name:        "multiple containers given and name not specified for 2nd contaniner",
+			command:     []string{"generate", "-f=" + Fixtures + "multi-containername/nginx.yaml"},
+			wantSuccess: false,
+			error:       "unable to perform controller operations: unable to fix data: unable to fix ControllerFields: unable to fix containers: please specify name for app.containers[1]",
+		},
+		{
+			name:    "multiple volume claims with same name",
+			command: []string{"generate", "-f=" + Fixtures + "multi-pvc-same-name/app.yaml"},
+			error:   `unable to perform controller operations: unable to validate data: unable to validate controller fields: error validating volume claims: duplicate entry of volume claim "foo"`,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			stdout, err := runCmd(t, tt.command)
+			if err != nil && !tt.wantSuccess {
+
+				if stdout != tt.error {
+					t.Fatalf("wanted error: %q\nbut got error: %q %v", tt.error, stdout, err)
+				} else {
+					t.Logf("failed with error: %q", stdout)
+					return
+				}
+			} else if err != nil && tt.wantSuccess {
+				t.Fatalf("wanted success, but test failed with error: %s %v", stdout, err)
+			} else if err == nil && !tt.wantSuccess {
+				t.Fatalf("expected to fail but passed")
+			}
+
+			// Read the data from the input file
+			data, err := ioutil.ReadFile(tt.input)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if diff := diff.Diff(string(data), fmt.Sprintf("%s\n", stdout)); diff != "" {
+				t.Fatalf("wanted: \n%s\n======================\ngot: \n%s"+
+					"\n======================\ndiff: %s", string(data), stdout, diff)
+			}
+
+		})
+	}
+}
+
+func runCmd(t *testing.T, args []string) (string, error) {
+	cmd := exec.Command("kedge", args...)
```

Comment:
might be good to make `kedge` a global variable just-in-case we port this code to another project (hint: kompose, haha)

Reply:
done